beetsplug/edit.py

```diff
@@ -0,0 +1,438 @@
+                        d[obj[field]] += (" " + str(obj['id']))
+                    a.append([field, [{f: i} for f, i in d.items()]])
+            return a
+        else:
+            for mod in objs:
+                a.append([{fi: mod[fi]}for fi in myfields])
+            return a
+
+    def get_all_fields(self, myfields, objs, opts):
+        a = []
+        if opts.sum:
+            fields = (library.Album.all_keys() if opts.album
+                      else library.Item.all_keys())
+            for field in sorted(fields):
+                # for every field get a dict
+                d = collections.defaultdict(str)
+                for obj in objs:
+                    # put all the ob-ids in the dict[field] as a string
+                    d[obj[field]] += (" " + str(obj['id']))
+                # for the field, we get the value and the users
+                a.append([field, [{f: i} for f, i in sorted(d.items())]])
+            return a
+        else:
+            for mod in objs:
+                a.append([{fi: mod[fi]} for fi in sorted(mod._fields)])
+            return a
+
+    def change_objs(self, dict_items):
+        # construct a yaml from the original object-fields
+        # and make a yaml that we can change in the text-editor
+        oldyaml = self.print_items[self.style](dict_items)
+        newyaml = self.print_items[self.style](dict_items)
+        new = NamedTemporaryFile(suffix='.yaml', delete=False)
+        new.write(newyaml)
+        new.close()
+        self.get_editor(new.name)
+
+        if ui.input_yn(ui.colorize('action_default', "done?(y)"), True):
+            while True:
+                try:
+                    with open(new.name) as f:
+                        newyaml = f.read()
+                        list(yaml.load_all(newyaml))
+                        break
+                except yaml.YAMLError as e:
+                    print_(ui.colorize('text_warning',
+                           "change this fault: {}".format(e)))
+                    print_("correct format for empty = - '' :")
+                    if ui.input_yn(
+                            ui.colorize('action_default', "fix?(y)"), True):
+                        self.get_editor(new.name)
+                        if ui.input_yn(ui.colorize(
+                           'action_default', "ok.fixed.(y)"), True):
+                            pass
+
+            return newyaml, oldyaml
+        else:
+            exit()
+
+    def get_editor(self, name):
+        if not self.ed:
+            editor = os.getenv('EDITOR')
+            if editor:
+                os.system(editor + " " + name)
+            else:
+                webbrowser.open(name, new=2, autoraise=True)
+        else:
+            callmethod = [self.ed]
+            if self.ed_args:
+                callmethod.extend(self.ed_args)
+            callmethod.append(name)
+            subprocess.check_call(callmethod)
+
+    def same_format(self, newset, opts):
```

Comment:
Sure.
2 different objects could come in the save method. 1 from sum 1 from your normal view. They have different structures so I make em look the same in same_format. I also make it print the title/artist instead of the Id. But if you delete the sum part you will probably not need that.